dali/core/mm/default_resource_test.cc

```diff
@@ -68,6 +68,36 @@ TEST(MMDefaultResource, GetResource_Pinned) {
     EXPECT_EQ(back_copy[i], static_cast<char>(i + 42));
 }
 
+
+TEST(MMDefaultResource, GetResource_Managed) {
+  auto *rsrc = GetDefaultResource<memory_kind::managed>();
+  ASSERT_NE(rsrc, nullptr);
+
+  CUDAStream stream = CUDAStream::Create(true);
+  char *mem = nullptr;
+  try {
+    mem = static_cast<char*>(rsrc->allocate(1000, 32));
+  } catch (const CUDAError &e) {
+    if ((e.is_drv_api() && e.drv_error() == CUDA_ERROR_NOT_SUPPORTED) ||
+        (e.is_rt_api() && e.rt_error() == cudaErrorNotSupported)) {
+      GTEST_SKIP() << "Unified memory not supported on this platform";
+    }
+  }
+  ASSERT_NE(mem, nullptr);
+  EXPECT_TRUE(mm::detail::is_aligned(mem, 32));
+  for (int i = 0; i < 1000; i++)
+    mem[i] = i + 42;
+  char back_copy[1000] = {};
+  CUDA_CALL(cudaMemcpyAsync(back_copy, mem, 1000, cudaMemcpyDeviceToHost, stream));
+  rsrc->deallocate_async(mem, 1000, 32, stream_view(stream));
+  CUDAEvent event = CUDAEvent::Create();
+  CUDA_CALL(cudaEventRecord(event, stream));
+  stream = CUDAStream();  // destroy the stream, it should still complete just fine
```

Comment:
Maybe `stream` could be inside a dedicated scope:
```
  char *mem = nullptr;
  char back_copy[1000] = {};
  {
    CUDAStream stream = CUDAStream::Create(true);
    try {
      mem = static_cast<char*>(rsrc->allocate(1000, 32));
    } catch (const CUDAError &e) {
      if ((e.is_drv_api() && e.drv_error() == CUDA_ERROR_NOT_SUPPORTED) ||
          (e.is_rt_api() && e.rt_error() == cudaErrorNotSupported)) {
        GTEST_SKIP() << "Unified memory not supported on this platform";
      }
    }
    ASSERT_NE(mem, nullptr);
    EXPECT_TRUE(mm::detail::is_aligned(mem, 32));
    for (int i = 0; i < 1000; i++)
      mem[i] = i + 42;
    CUDA_CALL(cudaMemcpyAsync(back_copy, mem, 1000, cudaMemcpyDeviceToHost, stream));
    rsrc->deallocate_async(mem, 1000, 32, stream_view(stream));
    CUDAEvent event = CUDAEvent::Create();
    CUDA_CALL(cudaEventRecord(event, stream));
  }
```

Reply:
Maybe... but this is only a cosmetic improvement and it would make this test stand out (there are more tests like this in this file) - reworking others would add unnecessary noise to this, already large, PR.